File: 2503-longest-subarray-with-maximum-bitwise-and/2503-longest-subarray-with-maximum-bitwise-and.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int x=*max_element(nums.begin(),nums.end());
        int n=nums.size();
        int res=INT_MIN;
        int count=1;
        for(int i=1;i<n;i++)
        {
            if(nums[i]==nums[i-1] && nums[i]==x)
            {
                count++;
            }
            else
            {
                res=max(res,count);
                count=1;
                
            }
        }
        
        res=max(res,count);
        return res;
    }
};

File: 2503-longest-subarray-with-maximum-bitwise-and/2503-longest-subarray-with-maximum-bitwise-and.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        mx = max(nums)
        count, res = 0, 0 
        for num in nums:
            if num==mx:
                count+=1
            else:
                res = max(res,count)
                count = 0
        res = max(res,count)
        return res